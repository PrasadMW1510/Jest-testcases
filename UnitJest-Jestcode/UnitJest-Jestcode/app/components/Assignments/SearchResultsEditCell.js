import React from 'react';
import PropTypes from 'prop-types';

const SearchResultsEditCell = ({ metaData, rowData, index, clickHandler }) => {
  const cell = (
    <div className="search-results-table__result-td-cell">
      <a
        id={rowData.assignment}
        role="link"
        tabIndex="-1"
        onClick={event => {
          clickHandler(event, rowData, index, metaData);
        }}
      >
        {rowData.assignment}
      </a>
    </div>
  );

  return cell;
};

SearchResultsEditCell.propTypes = {
  rowData: PropTypes.object,
  handleStudent: PropTypes.func,
};

export default SearchResultsEditCell;
