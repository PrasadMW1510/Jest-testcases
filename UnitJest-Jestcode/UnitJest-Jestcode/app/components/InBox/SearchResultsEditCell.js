import React from 'react';
import PropTypes from 'prop-types';
import btnM180 from 'images/gateway_assets/btn_M180.png';
import btnM180Y2 from 'images/gateway_assets/btn_M180Y2.png';
import btnfm from 'images/gateway_assets/btn_fm.png';
import btnR180NG from 'images/gateway_assets/btn_r180ng.png';
import btnS44 from 'images/gateway_assets/btn_s44.png';
import btniRead from 'images/gateway_assets/btniRead.png';
import rtngenabled from 'images/gateway_assets/rtng_enabled.jpg';

const SearchResultsEditCell = ({ rowData }) => {
  const cell = (
    <div className="search-results-table__result-td-cell">
      <button className="search-results-table__result-col-button">
        {String(rowData.communityId) === String('M180') &&
          String(rowData.assignment) === String('Simulation') && (
            <img className="inboxprogram-coursebutton" src={btnM180} alt="M180" />
          )}
        {String(rowData.communityId) === String('M180Y2') &&
          String(rowData.assignment) === String('Simulation') && (
            <img className="inboxprogram-coursebutton" src={btnM180Y2} alt="M180Y2" />
          )}
        {String(rowData.communityId) === String('M180') &&
          String(rowData.assignment) === String('mSkills Assessment') && (
            <img className="inboxprogram-coursebutton" src={btnfm} alt="mSkills Assessment" />
          )}
        {String(rowData.communityId) === String('M180Y2') &&
          String(rowData.assignment) === String('mSkills Assessment') && (
            <img className="inboxprogram-coursebutton" src={btnfm} alt="mSkills Assessment" />
          )}
        {String(rowData.communityId) === String('S44JR') && (
          <img className="inboxprogram-coursebutton" src={btniRead} alt="S44JR" />
        )}
        {String(rowData.communityId) === String('R180NG') && (
          <img className="inboxprogram-coursebutton" src={btnR180NG} alt="R180NG" />
        )}
        {String(rowData.communityId) === String('S44NG') && (
          <img className="inboxprogram-coursebutton" src={btnS44} alt="S44NG" />
        )}
        {String(rowData.communityId) === String('RTNG') && (
          <img className="inboxprogram-coursebutton" src={rtngenabled} alt="RTNG" />
        )}
      </button>
    </div>
  );

  return cell;
};

SearchResultsEditCell.propTypes = {
  rowData: PropTypes.object,
  handleStudent: PropTypes.func,
};

export default SearchResultsEditCell;
