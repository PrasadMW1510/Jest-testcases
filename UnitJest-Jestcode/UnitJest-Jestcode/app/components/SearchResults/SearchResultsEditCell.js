import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import myImage from 'images/Pencil.png';
import myDisk from 'images/disk.png';
import myNoDisk from 'images/no-disk.png';

const SearchResultsEditCell = ({ rowData, clickHandler }) => {
  const cell = (
    <div className="search-results-table__result-td-cell">
      {clickHandler === undefined ? (
        <Link to="#">{rowData.Title}</Link>
      ) : (
        <a
          id={rowData.Title && rowData.Title[0]}
          href="button"
          onClick={event => {
            clickHandler(event, rowData);
          }}
        >
          {rowData.Title}
        </a>
      )}
      <button className="search-results-table__result-col-button">
        {String(rowData.QuizActive) === String('true') &&
          String(rowData.QuizTeacherMade) === String('true') && (
            <img style={{ width: '18px' }} src={myImage} alt="SAM" />
          )}
        {String(rowData.QuizActive) === String('true') &&
          String(rowData.QuizTeacherMade) === String('false') && (
            <img style={{ width: '18px' }} src={myDisk} alt="SAM" />
          )}
        {String(rowData.QuizActive) === String('false') &&
          String(rowData.QuizTeacherMade) === String('false') && (
            <img style={{ width: '18px' }} src={myNoDisk} alt="SAM" />
          )}
        {String(rowData.QuizActive) === String('false') &&
          String(rowData.QuizTeacherMade) === String('true') && (
            <img style={{ width: '18px' }} src={myNoDisk} alt="SAM" />
          )}
      </button>
    </div>
  );

  return cell;
};

SearchResultsEditCell.propTypes = {
  rowData: PropTypes.object,
};

export default SearchResultsEditCell;
