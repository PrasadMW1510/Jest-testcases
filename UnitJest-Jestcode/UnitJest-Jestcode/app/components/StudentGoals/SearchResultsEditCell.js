import React from 'react';
import PropTypes from 'prop-types';

const SearchResultsEditCell = ({ metaData, row, clickHandler }) => (
  <div>
    <a
      role="link"
      tabIndex="-1"
      value={row.value}
      onClick={event => {
        clickHandler(event, row, metaData);
      }}
    >
      {' '}
      {row.value}
    </a>
  </div>
);

SearchResultsEditCell.propTypes = {
  row: PropTypes.object,
  metaData: PropTypes.object,
  clickHandler: PropTypes.func,
};

export default SearchResultsEditCell;
