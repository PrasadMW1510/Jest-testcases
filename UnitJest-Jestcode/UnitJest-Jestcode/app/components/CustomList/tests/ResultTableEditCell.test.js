import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchResultsEditCell from '../ResultTableEditCell';

describe('<AboutSamLink />', () => {
  let wrapper = null;
  let mockOnAboutClick = null;
  let mockItems = null;
  beforeEach(() => {
    mockItems = {
      name: 'SAM Server',
      version: 'v1.1',
    };
    mockOnAboutClick = jest.fn();
    wrapper = shallow(
      <SearchResultsEditCell rowData={mockItems} onRemoveClick={mockOnAboutClick} />
    );
  });

  it('Should render as expected', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('handleshowMeClick', () => {
    it('on click', () => {
      const clickButton = wrapper.find('.search-results-table__result-td-cell button');
      wrapper
        .find('.search-results-table__result-td-cell button')
        .simulate('click', { preventDefault() {} });
      expect(clickButton.exists()).toBeTruthy();
    });
  });
});
