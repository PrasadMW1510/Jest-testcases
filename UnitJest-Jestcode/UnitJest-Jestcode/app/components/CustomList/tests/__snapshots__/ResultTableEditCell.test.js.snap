// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutSamLink /> Should render as expected 1`] = `
<div
  className="search-results-table__result-td-cell"
>
  <button
    className="custom-table-remove-button"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
